### PR TITLE
Update support for S4 and enable regression testing

### LINKED ISSUE #653: Update support for S4
Support for S4 should be updated by building against Intel 2022.1 and the new hpc-stack suite on that system.  Additionally, regression testing support should be added.  @DavidHuber-NOAA will take on this task.

### PR DESCRIPTION
**Pull request by DavidHuber-NOAA**
## DESCRIPTION OF CHANGES: 
This both updates the module file and adds regression testing support for S4.  The hpc-stack library suite was recently built on S4 using the Intel 2022.1 compilers, which the module file now reflects.

## TESTS CONDUCTED: 
Regression test data were transferred to S4 and regression testing was performed.  All tests passed.

## DEPENDENCIES:
N/A

## DOCUMENTATION:
N/A

## ISSUE: 
Fixes #653

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
@JeffBeck-NOAA and @LarissaReames-NOAA do either of you have an 's4' account?

**Comment by JeffBeck-NOAA** [comment]
> @JeffBeck-NOAA and @LarissaReames-NOAA do either of you have an 's4' account?

Unfortunately, no.

**Comment by DavidHuber-NOAA** [comment]
@GeorgeGayno-NOAA If you need someone to perform regular regression testing on S4, I can commit to doing that once a month.  It is a tier 2 machine, so I don't believe it be regression tested for every PR.

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA If you need someone to perform regular regression testing on S4, I can commit to doing that once a month. It is a tier 2 machine, so I don't believe it be regression tested for every PR.

That is up to you. Since it is a tier 2 machine (and the repo managers don't have accounts there) we can't formally support it. So if you use that machine, running the tests periodically is a good idea.

File: modulefiles/build.s4.intel.lua [hunk 1]
```diff
@@ -0,0 +1,59 @@
+help([[
+Load environment to compile UFS_UTILS on Hera using Intel
```
Comment (GeorgeGayno-NOAA) [inline modulefiles/build.s4.intel.lua]: on S4

File: reg_tests/grid_gen/driver.s4.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+#-----------------------------------------------------------------------------
+#
+# Run grid generation consistency tests on S4.
+#
+# Set WORK_DIR to your working directory. Set the PROJECT_CODE and QUEUE
+# as appropriate.  To see which projects you are authorized to use,
+# type "account_params".
```
Comment (GeorgeGayno-NOAA) [inline reg_tests/grid_gen/driver.s4.sh]: Do you mean 'sacctmgr'?

File: reg_tests/snow2mdl/driver.s4.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+#-----------------------------------------------------------------------------
+#
+# Run snow2mdl consistency test on S4.
+#
+# Set $DATA_ROOT to your working directory.  Set the project code
+# and queue as appropriate.
+#
+# Invoke the script as follows:  sbatch $script
```
Comment (GeorgeGayno-NOAA) [inline reg_tests/snow2mdl/driver.s4.sh]: ./$script

**Review by GeorgeGayno-NOAA — APPROVED** [review]
You had a couple of typos. Otherwise, looks good.

File: modulefiles/build.s4.intel.lua (hunk 1 above)
Comment (DavidHuber-NOAA) [reply modulefiles/build.s4.intel.lua]: Fixed

File: reg_tests/grid_gen/driver.s4.sh (hunk 2 above)
Comment (DavidHuber-NOAA) [reply reg_tests/grid_gen/driver.s4.sh]: Yep, thanks for the catch

File: reg_tests/snow2mdl/driver.s4.sh (hunk 3 above)
Comment (DavidHuber-NOAA) [reply reg_tests/snow2mdl/driver.s4.sh]: Fixed, thanks.

**Comment by DavidHuber-NOAA** [comment]
@GeorgeGayno-NOAA Thanks for the quick review!

**Pull request merged**